lldb/source/Plugins/SymbolFile/DWARF/DWARFDIE.cpp

```diff
@@ -377,11 +378,6 @@ static void GetDeclContextImpl(DWARFDIE die,
       die = spec;
       continue;
     }
-    // To find the name of a type in a type unit, we must follow the signature.
-    if (DWARFDIE spec = die.GetReferencedDIE(DW_AT_signature)) {
```

Comment:
Could you elaborate on why we don't need this anymore?

Reply:
Because die.GetName() will now find the correct name on its own. And (unlike with DW_AT_specification) the correct context can be reconstructed by following the parent chain in the original dwarf unit. For nested structs like `Outer::Middle::Inner` we get something like this in the main dwarf unit:
```
0x00000029:   DW_TAG_structure_type
                DW_AT_declaration	(true)
                DW_AT_signature	(0xd351fbfc6a4cee7c) => points to Outer

0x00000032:     DW_TAG_structure_type
                  DW_AT_declaration	(true)
                  DW_AT_signature	(0x5f726fb54ccb6c95) => points to Outer::Middle

0x0000003b:       DW_TAG_structure_type
                    DW_AT_declaration	(true)
                    DW_AT_signature	(0xd3cee531644665ec) => points to Outer::Middle::Inner
```

Reply:
Ahh right, makes sense.

So for `DW_AT_specification`s, the referring DIE doesn't have the parent chain structure. So we *must* follow the DIE it referred to in order to reconstruct the context. With `DW_AT_signature` the context structure is maintained, so this now becomes redundant.